sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -283,81 +282,136 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
         throw new Error("Number of unsuccessful weeks is empty.");
       }
       const totalExistingUnsuccessfulWeeks =
         await this.getTotalFullTimeUnsuccessfulWeeks(studentId);
-
       // When total number of unsuccessful weeks hits minimum 68, add SSR restriction.
       if (
         totalExistingUnsuccessfulWeeks +
           scholasticStandingData.numberOfUnsuccessfulWeeks >=
         SCHOLASTIC_STANDING_MINIMUM_UNSUCCESSFUL_WEEKS
       ) {
-        return this.studentRestrictionSharedService.createRestrictionToSave(
-          studentId,
-          RestrictionCode.SSR,
-          auditUserId,
-          applicationId,
-        );
+        const ssrRestriction =
+          await this.studentRestrictionSharedService.createRestrictionToSave(
+            studentId,
+            RestrictionCode.SSR,
+            auditUserId,
+            applicationId,
+          );
+        restrictions.push(ssrRestriction);
+        return restrictions;
       }
     }
     if (
       scholasticStandingData.scholasticStandingChangeType ===
       StudentScholasticStandingChangeType.StudentWithdrewFromProgram
     ) {
+      // Check for "WTHD" restriction since it should always be created if one is not present and active.
+      const hasActiveWTHD = existingRestrictions.some(
+        (studentRestriction) =>
+          studentRestriction.restriction.restrictionCode ===
+            RestrictionCode.WTHD && studentRestriction.isActive,
+      );
       // Check if "WTHD" restriction is already present for the student,
-      // if not add "WTHD" restriction else add "SSR" restriction.
-      const isWTHDAlreadyExists =
-        await this.studentRestrictionService.hasAnyActiveRestriction(
-          studentId,
-          [RestrictionCode.WTHD],
-        );
-
-      const restrictionCode = isWTHDAlreadyExists
-        ? RestrictionCode.SSR
-        : RestrictionCode.WTHD;
-
-      return this.studentRestrictionSharedService.createRestrictionToSave(
-        studentId,
-        restrictionCode,
-        auditUserId,
-        applicationId,
+      // if not, then create a new one.
+      if (!hasActiveWTHD) {
+        const wthdRestriction =
+          await this.studentRestrictionSharedService.createRestrictionToSave(
+            studentId,
+            RestrictionCode.WTHD,
+            auditUserId,
+            applicationId,
+          );
+        restrictions.push(wthdRestriction);
+      }
+      // Check if an SSR or SSRN was created for the student.
+      const createdSSROrSSRN = restrictions.some((studentRestriction) =>
+        [RestrictionCode.SSR, RestrictionCode.SSRN].includes(
+          studentRestriction.restriction.restrictionCode as RestrictionCode,
+        ),
       );
+      if (hasActiveWTHD && !createdSSROrSSRN) {
+        // If the student does not have SSR or SSRN restriction created yet and there was
+        // already an active WTHD, then create a new SSR restriction.
+        const ssrRestriction =
```

Comment:
Can the pertinent logic of  UC and withdrawal be in a private method? otherwise, I felt it is difficult to understand the logic?

Reply:
It can, but I was not willing to extract this logic into a private method since I did not see it as a worth/good improvement. I would rather do a proper refactor, isolating the logic for each scholastic and creating new services to deal with them. That was what I mentioned in the chat and exposed to the Team.